hg/hg_test.go

```diff
@@ -27,6 +27,11 @@ var _ = Describe("Hg", func() {
 			Expect(unionOfPaths(repo.IncludePaths)).To(Equal("file('re:/path/1')|file('re:/path/2')|file('re:/path/3')"))
 		})
 
+		It("produces the correct branch fragment", func() {
+			Expect(repo.makeBranchQueryFragment()).To(Equal("literal:re:a_branch"))
+			Expect(emptyRepo.makeBranchQueryFragment()).To(Equal(""))
```

Comment:
`default` is currently set in [check][2], [in][1] and [out][3], so it wouldn't be empty string when it isn't defined. I didn't think that there was a need to alter the test data, which was against an empty `Repository`. I could make that change that however I've removed `makeBranchQueryFragment` as requested and so there isn't a need for this test anymore.

[1]: https://github.com/concourse/hg-resource/blob/master/hgresource/in.go#L30
[2]: https://github.com/concourse/hg-resource/blob/master/hgresource/check.go#L28
[3]: https://github.com/concourse/hg-resource/blob/master/hgresource/out.go#L210